src/modules/sync/lib/convert-tree-to-sync-options.ts

```diff
@@ -37,45 +143,32 @@ const getCommonNodes = ( tree: TreeNode[] ) => {
 
 export const convertTreeToPushOptions = ( tree: TreeNode[] ): PushOptionsWithSelections => {
 	const optionsToSync: SyncOption[] = [];
-	let specificSelections: PushOptionsWithSelections[ 'specificSelections' ] = undefined;
 
 	const isAll = tree.every( ( node ) => node.checked );
 	if ( isAll ) {
 		optionsToSync.push( SYNC_OPTIONS.all );
-	} else {
-		const { isDatabaseSelected, wpContent } = getCommonNodes( tree );
+		return { optionsToSync, specificSelections: undefined };
+	}
 
-		if ( isDatabaseSelected?.checked ) {
-			optionsToSync.push( SYNC_OPTIONS.sqls );
-		}
+	const { isDatabaseSelected, wpContent } = getCommonNodes( tree );
 
-		const wpContentChildren = wpContent?.children || [];
-		wpContentChildren.forEach( ( item ) => {
-			if ( ! isSyncOption( item.id ) ) {
-				return;
-			}
+	if ( isDatabaseSelected?.checked ) {
+		optionsToSync.push( SYNC_OPTIONS.sqls );
+	}
 
-			if ( item.checked || item.indeterminate ) {
-				optionsToSync.push( item.id );
-			}
+	let specificSelections: PushOptionsWithSelections[ 'specificSelections' ] = undefined;
 
-			if (
-				item.children &&
-				[ SYNC_OPTIONS.plugins, SYNC_OPTIONS.themes, SYNC_OPTIONS.uploads ].includes(
-					item.id as 'plugins' | 'themes' | 'uploads'
-				)
-			) {
-				const selectedItems = item.children
-					.filter( ( child ) => child.checked )
-					.map( ( child ) => child.name );
-				if ( selectedItems.length > 0 && selectedItems.length < item.children.length ) {
-					specificSelections = {
-						...specificSelections,
-						[ item.id ]: selectedItems,
-					};
-				}
+	if ( wpContent?.children?.length ) {
+		const selectedPaths = collectSelectedPaths( wpContent.children );
+		const syncOptionGroups = groupPathsBySyncOption( selectedPaths );
+
+		PATH_SYNC_MAP.forEach( ( { option } ) => {
+			if ( syncOptionGroups[ option ] ) {
+				optionsToSync.push( option );
 			}
 		} );
```

Comment:
```suggestion
		const selectedPaths = collectSelectedPaths( wpContent.children );

		PATH_SYNC_MAP.forEach( ( { option, pattern } ) => {
			const patternExistsAmongPaths = selectedPaths.some( ( path ) => {
				const normalizedPath = normalizePath( path );
				return pattern.test( normalizedPath );
			} );

			if ( patternExistsAmongPaths ) {
				optionsToSync.push( option );
			}
		} );
```
I found the logic in this file a little hard to follow at times, and I think we can simplify it along these lines. I haven't accounted for Unix/Windows-style paths here. The `normalizePath` function seems like a good place to do that. 

Making this change would allow us to remove the `getPathSyncOption` and `groupPathsBySyncOption` functions.

Reply:
Nice simplification. I applied this suggestion and cleaned up the unused functions in https://github.com/Automattic/studio/pull/1940/commits/a9cfee8b228b145ac8b45c41b02ccb61a976b48b